### ISSUE TITLE
Incorrect grad on complex number

### ISSUE THREAD

**Issue by mrlazy1708** [google/jax#15872]
### Description

Hi, thanks for maintaining JAX. I believe the behavior of taking gradient on the imag part of complex-valued input is incorrect, differing in sign. Here's an example

```python
import jax
import jax.numpy as np

jax.grad(lambda x: x.imag**2)(np.array(2j))
```

The above code gives `Array(0.-4.j, dtype=complex64)`. However considering the definition of grad, it should be positive. Here's a torch version of it, which gives the correct grad `(tensor(0.+4.j),)`:

```python
import torch

x = torch.tensor(2j, requires_grad=True)
y = x.imag**2
torch.autograd.grad(y, x)
```

### What jax/jaxlib version are you using?

jax/jaxlib v0.4.5 on GPU, v0.4.8 on CPU

### Which accelerator(s) are you using?

Tried on both CPU/GPU

### Additional system info

Python 3.8. Tried on both Ubuntu 22.04 LTS (x86) & MacOS 13.0

### NVIDIA GPU info

```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
```

**Comment by mattjj** [comment]
Thanks for raising this!

Actually this is just a different convention for what `grad` means; `torch` and `jax` follow different conventions, which differ by a complex conjugate. In JAX we _define_ `grad(f)(x) == jax.vjp(f, x)[1](1.0)[0]` (following a choice we made in [the original Autograd](https://github.com/hips/autograd), basically from at the time preferring `grad(lambda x: c * x) == c` for complex `c`). That means for complex inputs it does _not_ correspond to what you'd get from the more natural mathematical definitions, e.g. defining the gradient as the representer for the vjp-with-ones covector in terms of the standard (complex) inner product; instead it differs by a conjugate.

I'd like to switch conventions to the more mathematically motivated one (i.e. the same as what `torch` does). See #4891. But it just hasn't been a high priority. At the end of the day, it's just a convention, and you can just apply `jnp.conjugate`, or define your own `grad` function in terms of `jax.vjp`. (Notice, interestingly, that `jax.vjp` is unambiguous here: there's no need to choose some convention for what it means!)

The current JAX convention is [documented in the autodiff cookbook](https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#complex-numbers-and-differentiation).

Maybe we can keep discussion about changing the convention on the thread in #4891? But until we do, just apply a conjugate (or call `jax.vjp` directly).

**Comment by mrlazy1708** [comment]
Hi mattjj,

Thanks for prompt response. I encountered this when I was performing regression on complex inputs/outputs, and I was surprised that I can't directly minimize loss with jax/flax. To me it's a little bit counter-intuitive that I should `.conj` before gradient descent. I'll keep up with the thread you mentioned later, thanks!
